xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.cpp

```diff
@@ -0,0 +1,550 @@
+  if (flags & CONF_FLAGS_COLPRI_240M) return CMS_PRIMARIES_240M;
+  return CMS_PRIMARIES_BT709; // default to bt.709
+}
+
+bool CColorManager::GetVideo3dLut(int videoFlags, int *cmsToken, int *clutSize, uint16_t **clutData)
+{
+  CMS_PRIMARIES videoPrimaries = videoFlagsToPrimaries(videoFlags);
+  CLog::Log(LOGDEBUG, "video primaries: %d\n", (int)videoPrimaries);
+  switch (CSettings::GetInstance().GetInt("videoscreen.cmsmode"))
+  {
+  case CMS_MODE_3DLUT:
+    CLog::Log(LOGDEBUG, "ColorManager: CMS_MODE_3DLUT\n");
+    cur3dlutFile = CSettings::GetInstance().GetString("videoscreen.cms3dlut");
+    if (!Load3dLut(cur3dlutFile, clutData, clutSize))
+      return false;
+    curCmsMode = CMS_MODE_3DLUT;
+    break;
+
+  case CMS_MODE_PROFILE:
+    CLog::Log(LOGDEBUG, "ColorManager: CMS_MODE_PROFILE\n");
+#if defined(HAVE_LCMS2)
+    {
+      bool changed = false;
+      // check if display profile is not loaded, or has changed
+      if (curIccProfile != CSettings::GetInstance().GetString("videoscreen.displayprofile"))
+      {
+        changed = true;
+        // free old profile if there is one
+        if (m_hProfile)
+          cmsCloseProfile(m_hProfile);
+        // load profile
+        m_hProfile = LoadIccDisplayProfile(CSettings::GetInstance().GetString("videoscreen.displayprofile"));
+        if (!m_hProfile)
+          return false;
+        // detect blackpoint
+        if (cmsDetectBlackPoint(&m_blackPoint, m_hProfile, INTENT_PERCEPTUAL, 0))
+        {
+          CLog::Log(LOGDEBUG, "black point: %f\n", m_blackPoint.Y);
+        }
+        curIccProfile = CSettings::GetInstance().GetString("videoscreen.displayprofile");
+      }
+      // create gamma curve
+      cmsToneCurve* gammaCurve;
+      curIccGammaMode = (CMS_TRC_TYPE)CSettings::GetInstance().GetInt("videoscreen.cmsgammamode");
+      curIccGamma = CSettings::GetInstance().GetInt("videoscreen.cmsgamma");
+      gammaCurve =
+        CreateToneCurve(curIccGammaMode, curIccGamma/100.0f, m_blackPoint);
+
+      // create source profile
+      curIccWhitePoint = (CMS_WHITEPOINT)CSettings::GetInstance().GetInt("videoscreen.cmswhitepoint");
+      curIccPrimaries = (CMS_PRIMARIES)CSettings::GetInstance().GetInt("videoscreen.cmsprimaries");
+      CLog::Log(LOGDEBUG, "primaries setting: %d\n", (int)curIccPrimaries);
+      if (curIccPrimaries == CMS_PRIMARIES_AUTO) curIccPrimaries = videoPrimaries;
+      CLog::Log(LOGDEBUG, "source profile primaries: %d\n", (int)curIccPrimaries);
+      cmsHPROFILE sourceProfile =
+        CreateSourceProfile(curIccPrimaries, gammaCurve, curIccWhitePoint);
+
+      // link profiles
+      // TODO: intent selection, switch output to 16 bits?
+      cmsSetAdaptationState(0.0);
+#ifdef _DEBUG
+      int64_t start;
+      start = CurrentHostCounter();
+#endif
+      cmsHTRANSFORM deviceLink =
+        cmsCreateTransform(sourceProfile, TYPE_RGB_FLT,
+            m_hProfile, TYPE_RGB_FLT,
+            INTENT_ABSOLUTE_COLORIMETRIC, 0);
+#ifdef _DEBUG
```

Comment:
please drop those blocks. this won't ever get used. devs who are interested can add the measurements